src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -230,7 +230,18 @@ def dependencies(
         if entry is None:
             raise self._coordinate_not_found(key, coord)
 
-        return (entry, tuple(entries[(i.group, i.artifact)] for i in entry.dependencies))
+        dependency_entries = []
+        for d in entry.dependencies:
+            dependency = (d.group, d.artifact)
+            if dependency not in entries:
+                if d.packaging == "pom":
+                    # https://github.com/pantsbuild/pants/issues/20162#issuecomment-1813374036
```

Comment:
Preferably the codebase has at least a short standalone explanation, although you can include a link too for more details.

(And, BTW, Pants' convention is to use proper full sentences for comments.)